packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2111,6 +2115,14 @@ function commitMutationEffects(root: FiberRoot, renderPriorityLevel) {
       }
       case Deletion: {
         commitDeletion(root, nextEffect, renderPriorityLevel);
+        // If the effect has a nextEffect, add it to our list
+        if (nextEffect.nextEffect !== null) {
```

Comment:
Does this check do anything? Seems like it would leak if this was last in the list. The alternate could still have things to clear.

Reply:
That's a good point, let me try your other suggestion first.